jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -340,21 +341,27 @@ class DefaultAwsResourceCache(
 
     private fun runCacheMaintenance() {
         try {
-            var totalWeight = 0
-            val entries = cache.entries.asSequence().filter { it.value.value.isCompleted }.onEach { totalWeight += it.value.weight }.toList()
-            var exceededWeight = totalWeight - maximumCacheEntries
-            if (exceededWeight <= 0) return
-            entries.sortedBy { it.value.expiry }.forEach { (key, value) ->
-                if (exceededWeight <= 0) return@runCacheMaintenance
-                if (cache.computeRemoveIf(key) { it === value }) {
-                    exceededWeight -= value.weight
-                }
-            }
+            doRunCacheMaintenance()
         } finally {
             scheduleCacheMaintenance()
         }
     }
 
+    @VisibleForTesting
```

Comment:
I do hate this - but for internal state like this I just can't justify injection ... I'm very conflicted about it though

Reply:
i think we can lint for this if that makes you feel better

Reply:
lint for what? that it's only used in tests?

Reply:
yup, we have a number of `@VisibleForTesting` peppered around the code base but i'm not actually sure those are being respected